static/app/views/eventsV2/table/columnEditCollection.tsx

```diff
@@ -136,22 +142,71 @@ class ColumnEditCollection extends React.Component<Props, State> {
     this.props.onChange([...this.props.columns, newColumn]);
   };
 
-  handleUpdateColumn = (index: number, column: Column) => {
+  handleUpdateColumn = (index: number, updatedColumn: Column) => {
     const newColumns = [...this.props.columns];
-    if (column.kind === 'equation') {
+    if (updatedColumn.kind === 'equation') {
       this.setState(prevState => {
         const error = new Map(prevState.error);
-        error.set(index, parseArithmetic(column.field).error);
+        error.set(index, parseArithmetic(updatedColumn.field).error);
         return {
           ...prevState,
           error,
         };
       });
+    } else {
+      // Update any equations that contain the existing column
+      this.updateEquationFields(newColumns, index, updatedColumn);
     }
-    newColumns.splice(index, 1, column);
+    newColumns.splice(index, 1, updatedColumn);
     this.props.onChange(newColumns);
   };
 
+  updateEquationFields = (newColumns: Column[], index: number, updatedColumn: Column) => {
+    const oldColumn = newColumns[index];
+    const existingColumn = generateFieldAsString(newColumns[index]);
+    const updatedColumnString = generateFieldAsString(updatedColumn);
+    if (hasDuplicate(newColumns, oldColumn) || !isEquationColumn(updatedColumn)) {
```

Comment:
Nit: reorder these conditions since the second conditions should be faster to check